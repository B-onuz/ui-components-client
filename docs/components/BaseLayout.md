# BaseLayout (Component)

O Componente BaseLayout Ã© usado em conjunto com o componente `Sidebar`

## Como utilizar

```js
import BaseLayout from 'ui-components/lib/components/BaseLayout'
```

```jsx
import React, { useState } from 'react'
import BaseLayout from 'ui-components/lib/components/BaseLayout'
import Sidebar from 'ui-components/lib/components/Sidebar'

const DashboardSidebar = ({ ...rest }) => {
  return (
    <SidebarMenu {...rest} logo={<Logo width={180} my={3} src={logoImg} alt={'B-onuz Logo - legal services'} />}>
      {menuItems.map((item) => {
        return <SidebarMenuItem component={'header'} icon={<FontAwesomeIcon icon={item.icon} />} children={item.text} />
      })}
    </SidebarMenu>
  )
}

const Home = () => {
  const [menuIsOpen, setMenuIsOpen] = useState(true)

  const toggleMenu = () => {
    setMenuIsOpen(!menuIsOpen)
  }

  const closeMenu = () => {
    setMenuIsOpen(false)
  }

  return (
    <BaseLayout
      title={'Home page'}
      menuIsOpen={menuIsOpen}
      handleChangeMenuIsOpen={toggleMenu}
      sidebar={<DashboardSidebar open={menuIsOpen} onClose={closeMenu} width={248} />}
    >
      <h1>Lorem ipsum dolor sit amet</h1>
    </BaseLayout>
  )
}
```

## Styled-system props

- layout https://styled-system.com/table#layout
