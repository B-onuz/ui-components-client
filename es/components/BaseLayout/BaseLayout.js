"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents2 = require("reflexbox/styled-components");

var _HamburguerButton = _interopRequireDefault(require("../HamburguerButton"));

var _NavBar = _interopRequireDefault(require("../NavBar"));

var _Header = _interopRequireDefault(require("../Header"));

var _ProfileBox = _interopRequireDefault(require("../ProfileBox"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/BaseLayout/BaseLayout.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  height: 100%;\n  width: 90%;\n  align-self: flex-end;\n  white-space: nowrap;\n  @media only screen and (max-width: 768px) {\n    width: auto;\n    align-self: center;\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  transition: margin-left 0.2s linear;\n  ", "\n  @media only screen and (max-width: 768px) {\n    overflow-x: hidden;\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  flex: 1;\n  padding: 42px;\n  overflow: auto;\n  max-width: 100%;\n  display: flex;\n  flex-direction: column;\n  @media only screen and (min-width: 1640px) {\n    max-width: 1400px;\n  }\n  @media only screen and (max-width: 700px) {\n    padding: 24px 24px 42px;\n    overflow: unset;\n    margin-top: 10vh;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  height: 100%;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledBaseLayout = (0, _styledComponents["default"])(_styledComponents2.Flex)(_templateObject(), _styledSystem.color);

var ChildrenWrapper = _styledComponents["default"].div(_templateObject2());

var LayoutContainer = (0, _styledComponents["default"])(_styledComponents2.Box)(_templateObject3(), function (_ref) {
  var menuIsOpen = _ref.menuIsOpen,
      menuWidth = _ref.menuWidth;
  return !menuIsOpen ? "\n    margin-left: 0;\n  " : "\n    @media only screen and (min-width: 1024px) {\n      margin-left: " + menuWidth + "px;\n    }\n  ";
});

var CornerContentWrapper = _styledComponents["default"].div(_templateObject4());

var BaseLayout = function BaseLayout(_ref2) {
  var children = _ref2.children,
      _ref2$menuIsOpen = _ref2.menuIsOpen,
      menuIsOpen = _ref2$menuIsOpen === void 0 ? false : _ref2$menuIsOpen,
      _ref2$sidebar = _ref2.sidebar,
      sidebar = _ref2$sidebar === void 0 ? null : _ref2$sidebar,
      _ref2$cornerContent = _ref2.cornerContent,
      cornerContent = _ref2$cornerContent === void 0 ? null : _ref2$cornerContent,
      _ref2$userInfo = _ref2.userInfo,
      userInfo = _ref2$userInfo === void 0 ? null : _ref2$userInfo,
      _ref2$menuWidth = _ref2.menuWidth,
      menuWidth = _ref2$menuWidth === void 0 ? 248 : _ref2$menuWidth,
      handleChangeMenuIsOpen = _ref2.handleChangeMenuIsOpen,
      title = _ref2.title,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children", "menuIsOpen", "sidebar", "cornerContent", "userInfo", "menuWidth", "handleChangeMenuIsOpen", "title"]);

  return /*#__PURE__*/_react["default"].createElement(StyledBaseLayout, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 73,
      columnNumber: 5
    }
  }), sidebar, /*#__PURE__*/_react["default"].createElement(LayoutContainer, {
    display: 'flex',
    flexDirection: 'column',
    flex: 1,
    menuWidth: menuWidth,
    menuIsOpen: menuIsOpen,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 75,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react["default"].createElement(_NavBar["default"], {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 76,
      columnNumber: 9
    }
  }, !!sidebar && /*#__PURE__*/_react["default"].createElement(_HamburguerButton["default"], {
    backgroundColor: '#f4f4f4',
    isOpen: !!sidebar ? menuIsOpen : false,
    onClick: handleChangeMenuIsOpen,
    color: 'primary',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 78,
      columnNumber: 13
    }
  }, !open ? 'Abrir menu' : 'Fechar menu'), /*#__PURE__*/_react["default"].createElement(_Header["default"], {
    title: title || '',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 82,
      columnNumber: 11
    }
  }), !!cornerContent && /*#__PURE__*/_react["default"].createElement(CornerContentWrapper, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 83,
      columnNumber: 31
    }
  }, cornerContent)), /*#__PURE__*/_react["default"].createElement(ChildrenWrapper, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 85,
      columnNumber: 9
    }
  }, children)));
};

BaseLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  children: _propTypes["default"].element.isRequired
} : {};
BaseLayout.defaultProps = {
  /*todo: defaultProps*/
};
var _default = BaseLayout;
exports["default"] = _default;