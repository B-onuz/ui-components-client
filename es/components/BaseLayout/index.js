"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _BaseLayout = _interopRequireDefault(require("./BaseLayout"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _BaseLayout["default"];
exports["default"] = _default;