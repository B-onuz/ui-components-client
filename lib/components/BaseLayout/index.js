"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _BaseLayout = _interopRequireDefault(require("./BaseLayout"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _BaseLayout["default"];
exports["default"] = _default;
module.exports = exports.default;