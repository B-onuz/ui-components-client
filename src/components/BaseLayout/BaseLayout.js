import React from 'react'
import styled from 'styled-components'
import { color } from 'styled-system'
import PropTypes from 'prop-types'
import { Flex, Box } from 'reflexbox/styled-components'
import HamburguerButton from '../HamburguerButton'
import Navbar from '../NavBar'
import Header from '../Header'
import ProfileBox from '../ProfileBox'

const StyledBaseLayout = styled(Flex)`
  ${color}
  height: 100%;
`

const ChildrenWrapper = styled.div`
  flex: 1;
  padding: 0px 42px 42px 42px;
  overflow: auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
  @media only screen and (min-width: 1640px) {
    max-width: 1400px;
  }
  @media only screen and (max-width: 600px) {
    padding: 24px 24px 42px;
    overflow: unset;
    margin-top: 10vh;
  }
`

const LayoutContainer = styled(Box)`
  transition: margin-left 0.2s linear;
  ${({ menuIsOpen, menuWidth }) =>
    !menuIsOpen
      ? `
    margin-left: 0;
  `
      : `
    @media only screen and (min-width: 1024px) {
      margin-left: ${menuWidth}px;
    }
  `}
  @media only screen and (max-width: 768px) {
    overflow-x: hidden;
  }
`

const CornerContentWrapper = styled.div`
  height: 100%;
  width: 95%;
  align-self: flex-end;
  white-space: nowrap;
  @media only screen and (max-width: 600px) {
    width: auto;
    align-self: center;
  }
`

const BaseLayout = ({
  children,
  menuIsOpen = false,
  sidebar = null,
  cornerContent = null,
  userInfo = null,
  menuWidth = 248,
  handleChangeMenuIsOpen,
  title,
  ...rest
}) => {
  return (
    <StyledBaseLayout {...rest}>
      {sidebar}
      <LayoutContainer display={'flex'} flexDirection={'column'} flex={1} menuWidth={menuWidth} menuIsOpen={menuIsOpen}>
        <Navbar>
          {!!sidebar && (
            <HamburguerButton backgroundColor={'#FFFFFF'} isOpen={!!sidebar ? menuIsOpen : false} onClick={handleChangeMenuIsOpen} color={'primary'}>
              {!open ? 'Abrir menu' : 'Fechar menu'}
            </HamburguerButton>
          )}
          <Header title={title || ''} />
          {!!cornerContent && <CornerContentWrapper>{cornerContent}</CornerContentWrapper>}
        </Navbar>
        <ChildrenWrapper>{children}</ChildrenWrapper>
      </LayoutContainer>
    </StyledBaseLayout>
  )
}

BaseLayout.propTypes = {
  children: PropTypes.element.isRequired,
}

BaseLayout.defaultProps = {
  /*todo: defaultProps*/
}

export default BaseLayout
