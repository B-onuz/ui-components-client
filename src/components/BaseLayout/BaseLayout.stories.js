import React, { useState } from 'react'
import BaseLayout from './BaseLayout'
import Fieldset from '../Fieldset'
import TextField from '../TextField'
import { Box, Flex } from 'reflexbox'
import SidebarMenu from '../SidebarMenu'
import SidebarMenuItem from '../SidebarMenuItem'
import CheckBox from '../CheckBox'
import Button from '../Button'
import ProfileBox from '../ProfileBox'
import bnz_icon from '../../assets/bnz_icon.png'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Logo from '../Logo'
import logoImg from '../../assets/logo_white.svg'
import { faHome, faDice, faCodeBranch, faAdjust, faKey, faKissBeam, faKissWinkHeart } from '@fortawesome/free-solid-svg-icons'

const menuItems = [
  { icon: faHome, text: 'Home' },
  { icon: faDice, text: 'Dados' },
  { icon: faCodeBranch, text: 'Branch' },
  { icon: faAdjust, text: 'Adjusts Darkmode' },
  { icon: faKey, text: 'Chaves de acesso' },
]

const DashboardSidebar = ({ ...rest }) => {
  return (
    <SidebarMenu {...rest} logo={<Logo width={180} my={3} src={logoImg} alt={'B-onuz Logo - legal services'} />}>
      {menuItems.map((item) => {
        return <SidebarMenuItem component={'header'} icon={<FontAwesomeIcon icon={item.icon} />} children={item.text} />
      })}
    </SidebarMenu>
  )
}

export default {
  title: 'BaseLayout',
  component: BaseLayout,
}

export const BaseLayoutExample = () => {
  return (
    <BaseLayout title="Cadastro de advogados">
      <h1>Layout base</h1>
    </BaseLayout>
  )
}

export const BaseLayoutWithSidebar = () => {
  const [menuIsOpen, setMenuIsOpen] = useState(true)

  const toggleMenu = () => {
    setMenuIsOpen(!menuIsOpen)
  }

  const closeMenu = () => {
    setMenuIsOpen(false)
  }
  return (
    <BaseLayout
      title={'Cadastro de advogados'}
      menuIsOpen={menuIsOpen}
      handleChangeMenuIsOpen={toggleMenu}
      cornerContent={<ProfileBox src={bnz_icon} balanceValue={2300} />}
      sidebar={<DashboardSidebar open={menuIsOpen} onClose={closeMenu} width={248} />}
    >
      <form noValidate>
        <Fieldset legend={'Informações pessoais'}>
          <Flex flexWrap={'wrap'} mx={-2}>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="name" name={'name'} label={'Nome'} m={2} placeholder={'Nome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="lastname" name={'lastname'} label={'Sobrenome'} m={2} placeholder={'Sobrenome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="fullname" name={'fullname'} label={'Exibir Como:'} m={2} placeholder={'Nome + Sobrenome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="name" name={'name'} label={'Nome'} m={2} placeholder={'Nome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="lastname" name={'lastname'} label={'Sobrenome'} m={2} placeholder={'Sobrenome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="fullname" name={'fullname'} label={'Exibir Como:'} m={2} placeholder={'Nome + Sobrenome'} />
            </Box>
          </Flex>
        </Fieldset>
        <Fieldset mt={4} legend={'Informações Profissionais'}>
          <Flex flexWrap={'wrap'} mx={-2}>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="name" name={'name'} label={'Nome'} m={2} placeholder={'Nome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="lastname" name={'lastname'} label={'Sobrenome'} m={2} placeholder={'Sobrenome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="fullname" name={'fullname'} label={'Exibir Como:'} m={2} placeholder={'Nome + Sobrenome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="name" name={'name'} label={'Nome'} m={2} placeholder={'Nome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="lastname" name={'lastname'} label={'Sobrenome'} m={2} placeholder={'Sobrenome'} />
            </Box>
            <Box width={[1, 1 / 2, 1 / 3]}>
              <TextField id="fullname" name={'fullname'} label={'Exibir Como:'} m={2} placeholder={'Nome + Sobrenome'} />
            </Box>

            <Box width={[1, 1, 2 / 3]}>
              <TextField id="fullname" name={'fullname'} label={'Exibir Como:'} m={2} placeholder={'Nome + Sobrenome'} />
            </Box>
          </Flex>
        </Fieldset>
        <Fieldset mt={4} legend={'Seleciona especialidades do advogado'}>
          <Flex m={-2} flexWrap={'wrap'} flexDirection={['column', 'column', 'row']}>
            <Box m={2}>
              <CheckBox id="societario" label={'Societário'} />
            </Box>
            <Box m={2}>
              <CheckBox id="propriedade-intelectual" label={'Propriedade Intelectual'} />
            </Box>
            <Box m={2}>
              <CheckBox id="tributario" label={'Tributário'} />
            </Box>
            <Box m={2}>
              <CheckBox id="contratos" label={'Contratos'} />
            </Box>
            <Box m={2}>
              <CheckBox id="investimentos" label={'Investimentos'} />
            </Box>
            <Box m={2}>
              <CheckBox id="contencioso" label={'Contencioso'} />
            </Box>
          </Flex>
        </Fieldset>
        <Fieldset mt={4} legend={'Administração da plataforma B-onuz'}>
          <CheckBox id="admin" label={'Adicionar esse advogado ao grupo de Administradores da Plataforma'} />
        </Fieldset>
        <Flex justifyContent={'flex-end'}>
          <Button m={2}>Cancelar</Button>
          <Button type="submit" color={'primary'} m={2}>
            Adicionar
          </Button>
        </Flex>
      </form>
    </BaseLayout>
  )
}

export const BaseLayoutCornerContentExample = () => {
  return <BaseLayout title="Cadastro de advogados" cornerContent={<ProfileBox src={bnz_icon} balanceValue={2300} />} />
}
