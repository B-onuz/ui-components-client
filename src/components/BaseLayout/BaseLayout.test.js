import React from 'react'
import BaseLayout from './BaseLayout.js'
import renderer from 'react-test-renderer'

test('should render BaseLayout without sidebar', () => {
  const component = renderer.create(
    <BaseLayout title={'Cadastro de advogados'}>
      <p>Lorem ipsum dolor sit amet</p>
    </BaseLayout>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
