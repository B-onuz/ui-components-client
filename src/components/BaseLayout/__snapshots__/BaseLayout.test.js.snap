// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BaseLayout without sidebar 1`] = `
<div
  className="sc-AxjAm sc-AxirZ sc-fzoyAV inmeFV"
>
  <div
    className="sc-AxjAm sc-fzoXzr fzTmwL"
    display="flex"
  >
    <div
      className="sc-AxhUy cPQMzN"
    >
      <header
        className="sc-Axmtr fiNXqW"
        fontSize={
          Array [
            2,
            3,
            4,
          ]
        }
      >
        <h1
          className="sc-AxgMl gJMJsZ"
          color="darkRed"
        >
          Cadastro de advogados
        </h1>
      </header>
    </div>
    <div
      className="sc-fzoLag djknVU"
    >
      <p>
        Lorem ipsum dolor sit amet
      </p>
    </div>
  </div>
</div>
`;
